compiler/ml/typecore.ml

```diff
@@ -2256,6 +2256,18 @@ let rec type_exp ~context ?recarg env sexp =
 *)
 
 and type_expect ~context ?in_function ?recarg env sexp ty_expected =
+  (* Special errors for braced identifiers passed to records *)
+  let context =
+    match sexp.pexp_desc with
+    | Pexp_ident _ ->
+      if
+        sexp.pexp_attributes
+        |> List.exists (fun (attr, _) -> attr.txt = "res.braces")
+        && is_record_type ~extract_concrete_typedecl ~env ty_expected
+      then Some Error_message_utils.BracedIdent
+      else context
+    | _ -> context
+  in
```

Comment:
@cristianoc I've tried to guard this lookup as much as possible, but do you think it's worth seeing if this processing could be moved to only happen if there's an actual type error? I guess I could catch and reraise.